include/ros_alarms/listener.hpp

```diff
@@ -94,11 +110,25 @@ ::AlarmListener(ros::NodeHandle &nh, std::string alarm_name)
 {
   // Service to query alarm server
   __get_alarm = __nh.serviceClient<ros_alarms::AlarmGet>("/alarm/get");
-  __get_alarm.waitForExistence();
-
+  bool service_exists = __get_alarm.waitForExistence(ros::Duration(2.0));
+  if(!service_exists)
+  {
+    __ok = false;
+    std::string msg = __PRETTY_FUNCTION__;
+    msg += ": timed out waiting for service /alarm/get";
+    ROS_WARN(msg.c_str());
+  }
 
   // Subscribes to list of recently modified alarms from alarm server
-  __update_subscriber = __nh.subscribe("/alarm/updates", 10, &AlarmListener<callable_t>::__alarm_update, this);
+  try
+  {
+    __update_subscriber = __nh.subscribe("/alarm/updates", 10, &AlarmListener<callable_t>::__alarm_update, this);
+  }
+  catch(ros::Exception &e)
+  {
+    __ok = false;
+    ROS_WARN((std::string(__PRETTY_FUNCTION__) + ": " + e.what()).c_str());
```

Comment:
Should it crash here (or up on line 119)? When a node crashes it can be set to automatically come back online which may be better than "silently" crashing (such that the messages could be part of a giant text wall).